src/fund/components/FundCardAmountInput.tsx

```diff
@@ -0,0 +1,167 @@
+import { type ChangeEvent, useCallback, useEffect, useRef } from 'react';
+import { cn, text } from '../../styles/theme';
+import type { FundCardAmountInputPropsReact } from '../types';
+import { formatDecimalInputValue } from '../utils/formatDecimalInputValue';
+import { truncateDecimalPlaces } from '../utils/truncateDecimalPlaces';
+import { FundCardCurrencyLabel } from './FundCardCurrencyLabel';
+
+export const FundCardAmountInput = ({
+  fiatValue,
+  setFiatValue,
+  cryptoValue,
+  setCryptoValue,
+  currencySign,
+  assetSymbol,
+  inputType = 'fiat',
+  exchangeRate = 1,
+}: FundCardAmountInputPropsReact) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const hiddenSpanRef = useRef<HTMLSpanElement>(null);
+  const currencySpanRef = useRef<HTMLSpanElement>(null);
+
+  const value = inputType === 'fiat' ? fiatValue : cryptoValue;
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      let value = e.target.value;
+
+      value = formatDecimalInputValue(value);
+
+      if (inputType === 'fiat') {
+        const fiatValue = truncateDecimalPlaces(value, 2);
+        setFiatValue(fiatValue);
+
+        // Calculate the crypto value based on the exchange rate
+        const calculatedCryptoValue = String(
+          Number(value) * Number(exchangeRate),
+        );
+
+        const resultCryptoValue = truncateDecimalPlaces(
+          calculatedCryptoValue,
+          8,
+        );
```

Comment:
nit: you could just use toFixed on your calculatedCryptoValue before casting it to a string, below as well for fiat